service/history/workflow/mutable_state_impl.go

```diff
@@ -1184,6 +1184,12 @@ func (ms *MutableStateImpl) DeletePendingSignal(
 func (ms *MutableStateImpl) writeEventToCache(
 	event *historypb.HistoryEvent,
 ) {
+
+	// skip adding to cache if event is too large
+	if event.Size() > ms.config.HistoryCacheEventMaxSize() {
+		return
+	}
```

Comment:
I think this check should be done by the eventsCache impl? Mutable state may not be the only caller of PutEvent (in the future)?